### PR TITLE
Get rid of occasional failures of TestSourcesTest.test_longStream

### LINKED ISSUE #2488: com.hazelcast.jet.pipeline.test.TestSourcesTest.test_longStream[0: mode=member]
_master_ (commit INSERT_COMMIT_ID_HERE)

Failed on Oracle JDK 11: http://jenkins.hazelcast.com/job/jet-oss-master-oracle-jdk11/269/testReport/com.hazelcast.jet.pipeline.test/TestSourcesTest/test_longStream_0__mode_member_/

Stacktrace:
```
java.lang.AssertionError: 

Expected: exception with message a string containing "com.hazelcast.jet.pipeline.test.AssertionCompletedException"
     but: message was "Exception in ProcessorTasklet{assertCollectedEventually#1}: java.lang.AssertionError: java.lang.AssertionError: Invalid items per second"
Stacktrace was: com.hazelcast.jet.JetException: Exception in ProcessorTasklet{assertCollectedEventually#1}: java.lang.AssertionError: java.lang.AssertionError: Invalid items per second
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:373)
	at java.base/java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:803)
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:346)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.AssertionError: java.lang.AssertionError: Invalid items per second
	at com.hazelcast.jet.pipeline.test.AssertionSinks$CollectingSinkWithTimer.timer(AssertionSinks.java:214)
	at com.hazelcast.function.ConsumerEx.accept(ConsumerEx.java:46)
	at com.hazelcast.jet.impl.pipeline.test.AssertionP.maybeFireTimer(AssertionP.java:67)
	at com.hazelcast.jet.impl.pipeline.test.AssertionP.tryProcess(AssertionP.java:60)
	at com.hazelcast.jet.impl.processor.ProcessorWrapper.tryProcess(ProcessorWrapper.java:103)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:283)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:268)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:242)
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:366)
	... 3 more
Caused by: java.lang.AssertionError: Invalid items per second
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.hazelcast.jet.pipeline.test.TestSourcesTest.lambda$test_longStream$d543e020$1(TestSourcesTest.java:64)
	at com.hazelcast.function.ConsumerEx.accept(ConsumerEx.java:46)
	at com.hazelcast.jet.pipeline.test.AssertionSinks$CollectingSinkWithTimer.receive(AssertionSinks.java:203)
	at com.hazelcast.function.BiConsumerEx.accept(BiConsumerEx.java:47)
	at com.hazelcast.jet.impl.pipeline.test.AssertionP.tryProcess(AssertionP.java:75)
	at com.hazelcast.jet.core.AbstractProcessor.tryProcess0(AbstractProcessor.java:187)
	at com.hazelcast.jet.core.AbstractProcessor.process0(AbstractProcessor.java:591)
	at com.hazelcast.jet.core.AbstractProcessor.process(AbstractProcessor.java:109)
	at com.hazelcast.jet.impl.processor.ProcessorWrapper.process(ProcessorWrapper.java:93)
	at com.hazelcast.jet.impl.pipeline.FunctionAdapter$AdaptingProcessor.process(FunctionAdapter.java:226)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.processInbox(ProcessorTasklet.java:416)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:291)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:286)
	... 6 more

	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.junit.rules.ExpectedException.handleException(ExpectedException.java:252)
	at org.junit.rules.ExpectedException.access$000(ExpectedException.java:106)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:241)
	at com.hazelcast.test.DumpBuildInfoOnFailureRule$1.evaluate(DumpBuildInfoOnFailureRule.java:37)
	at com.hazelcast.test.jitter.JitterRule$1.evaluate(JitterRule.java:104)
	at com.hazelcast.test.metrics.MetricsRule$1.evaluate(MetricsRule.java:62)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:50)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:29)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Standard output:
```
Hiccups measured while running test 'test_longStream[0: mode=member](com.hazelcast.jet.pipeline.test.TestSourcesTest):'
19:38:50, accumulated pauses: 679 ms, max pause: 144 ms, pauses over 1000 ms: 0
19:38:55, accumulated pauses: 483 ms, max pause: 41 ms, pauses over 1000 ms: 0
19:39:00, accumulated pauses: 50 ms, max pause: 10 ms, pauses over 1000 ms: 0
```

### PR DESCRIPTION
**Pull request by jbartok**
Fixes #2488 

First of all: `TestSources` is awesome! It was really hard to make it skip a beat. Finally managed to reproduce the problem with load on the system and a large local parallelism count for the source.

Made a change to make the cause of the failure more obvious (if it happens again). Tried to make it not happen again by waiting for more window results and ignoring not just the first one, but the first two (this is how I saw it fail every single time, by having the second window not be quite up to the items per second rate yet).

Can't reproduce it anymore.

### CODE REVIEW

File: hazelcast-jet-core/src/test/java/com/hazelcast/jet/pipeline/test/TestSourcesTest.java [hunk 1]
```diff
@@ -53,16 +55,16 @@ public void test_items() {
...
          .window(tumbling(SECONDS.toMillis(1)))
          .aggregate(counting())
-         .apply(assertCollectedEventually(timeoutSeconds, windowResults -> {
-             assertTrue("Didn't receive at least three results", windowResults.size() >= numWindowResultsToWaitFor);
-             assertTrue("Invalid items per second",
-                     windowResults.stream().skip(1).allMatch(wr -> wr.result() == itemsPerSecond));
+         .apply(assertCollectedEventually(timeoutSeconds, results -> {
+             assertTrue("Didn't receive at least three results", results.size() >= numWindowResultsToWaitFor);
+             assertTrue("Invalid items per second: " + results.stream().map(WindowResult::result).collect(toList()),
+                     results.stream().skip(2).allMatch(wr -> wr.result() == itemsPerSecond));
```
Comment (viliam-durina) [inline hazelcast-jet-core/src/test/java/com/hazelcast/jet/pipeline/test/TestSourcesTest.java:67]: Why skipping two instead of 1 helps?

Comment (jbartok) [reply hazelcast-jet-core/src/test/java/com/hazelcast/jet/pipeline/test/TestSourcesTest.java:67]: I don't exactly know, but that's how it was failing. I'm assuming that in the problematic high system load scenarios it just takes a bit longer for the source instances to get up to speed. I've never seen the 3rd or further windows not have the perfect number of items in them.

Comment (viliam-durina) [reply hazelcast-jet-core/src/test/java/com/hazelcast/jet/pipeline/test/TestSourcesTest.java:67]: "Seems to work"
:-)

Comment (jbartok) [reply hazelcast-jet-core/src/test/java/com/hazelcast/jet/pipeline/test/TestSourcesTest.java:67]: How is such a test supposed to be deterministic in the first place? How can a source be expected to produce a certain rate of events, regardless of how overloaded the machine running it is? So yes, the non-perfect test seems to be working a bit better like this.

**Comment by mtopolnik** [comment]
Same failure was addressed in #2483, closing this one.

**Pull request closed without merging**